l10n_ch_payment_slip/invoice.py

```diff
@@ -52,13 +53,11 @@ def get_bvr_ref(self, cr, uid, move_line_id, context=None):
 
         if not self._is_generate_bvr(cr, uid, move_line.invoice, context):
             return ''
-
-        reference = self._compute_bvr_ref(cr, uid, move_line, context)
-        if (move_line.transaction_ref and
-                move_line.transaction_ref != reference):
-            # the line has already a transaction id and it is not
-            # a BVR reference
-            return ''
+        if move_line.transaction_ref:
+            return move_line.transaction_ref
```

Comment:
The BVR report  call the get_bvr_ref without an sequence_int parameter (I cannot determine the sequence in the report), so if the move line have already a transaction_ref I simply return it.
The test at line #L54 already implement a test, if it's a BVR or not, so at #L56 we have a valid transaction_ref.